compal/__init__.py

```diff
@@ -926,6 +927,204 @@ def turn_off(self, debug=False):
+            ))
+
+        return all_interfaces
+
+    @property
+    def wifi_guest_network_settings(self):
+        """
+        Read the wifi guest network settings
+        """
+        xml = self.wifi_guest_network_settings_xml
+        guest_networks_2g = WifiGuestNetworkSettings.band_guest_networks(xml, "2g")
+        guest_networks_5g = WifiGuestNetworkSettings.band_guest_networks(xml, "5g")
+
+        return GuestNetworkSettings(
+            guest_networks_2g,
+            guest_networks_5g,
+        )
+
+    @staticmethod
+    def __compare_wifi_settings(old_settings, new_settings, interface_index, changes):
+        """
+        Compare two settings objects for changes and
+        return:
+            bln_changes: True if there were changes
+            changes: Python dict that contains the changes
+        """
+
+        def iterate_changes(old_settings, new_settings, changes):
+            for attr, old_value in old_settings.__dict__.items():
+                if isinstance(old_value, BandSetting):
+                    changes.update({attr: {}})
+                    iterate_changes(
+                        getattr(old_settings, attr),
+                        getattr(new_settings, attr),
+                        changes[attr],
+                    )
+                else:
+                    new_value = getattr(new_settings, attr)
+                    if old_value != new_value:
+                        changes.update({attr: f"{old_value} -> {new_value}"})
+                        bln_changes[0] = True
+
+        changes = {}
+        bln_changes = [False]
+        iterate_changes(old_settings.guest_networks_2g[interface_index],
+                        new_settings.guest_networks_2g[interface_index], changes)
+        iterate_changes(old_settings.guest_networks_5g[interface_index],
+                        new_settings.guest_networks_5g[interface_index], changes)
+        return bln_changes[0], changes
+
+    def __check_router_status(self, new_guest_network_settings, interface_index, debug):
+        """
+        Checks if (1) the new user settings were changed in the router (via checking
+        the getter fun:307 [WIRELESSGUESTNETWORK] in the router-xml-file), (2) prints out a
+        progress bar and (3) prints out if the process was successful.
+        """
+        router_settings = None
+
+        progress = 0
+
+        def print_progress_bar(progress, total, postfix=""):
+            progress = total if progress > total else progress
+            per_progress = int(progress / total * 100)
+            postfix = f"[{postfix}]" if postfix != "" else ""
+            print(
+                f"\r|{'█' * progress}{'-' * (total - progress)}| {per_progress}%, {progress}sec\t{postfix}",
+                end="",
+            )
+
+        total = 24
+        bln_changes = True
+        start_time = time.time()
+        changes = ""
+        if debug:
+            print(f"\n--- WAITING FOR ROUTER TO UPDATE ---")
```

Comment:
```suggestion
            print("\n--- WAITING FOR ROUTER TO UPDATE ---")
```

The linter only accepts string-interpolation-syntax (`f"..."`) if there is a variable being used. So `f"{total}"` is ok but `f"not using a variable"` would not be accepted.

Reply:
Thanks, it is fixed.